From: Philipp Zabel <p.zabel@pengutronix.de>
Date: Thu, 6 Aug 2015 10:43:43 +0200
Subject: [PATCH] gpu: drm/panel: Add DLC DLC0700YZG-1 panel (WIP)

This patch adds support for DLC DLC0700YZG-1 1024x600 LVDS panels
to the simple-panel driver.

Signed-off-by: Philipp Zabel <p.zabel@pengutronix.de>
---
 .../devicetree/bindings/panel/dlc,dlc0700yzg-1.txt |  7 ++++++
 .../devicetree/bindings/vendor-prefixes.txt        |  1 +
 drivers/gpu/drm/panel/panel-simple.c               | 28 ++++++++++++++++++++++
 3 files changed, 36 insertions(+)
 create mode 100644 Documentation/devicetree/bindings/panel/dlc,dlc0700yzg-1.txt

diff --git a/Documentation/devicetree/bindings/panel/dlc,dlc0700yzg-1.txt b/Documentation/devicetree/bindings/panel/dlc,dlc0700yzg-1.txt
new file mode 100644
index 000000000000..30ad3087542f
--- /dev/null
+++ b/Documentation/devicetree/bindings/panel/dlc,dlc0700yzg-1.txt
@@ -0,0 +1,7 @@
+DLC Display Co. DLC0700YZG-1 7.0" WSVGA TFT LCD panel
+
+Required properties:
+- compatible: should be "dlc,dlc070yzg-1"
+
+This binding is compatible with the simple-panel binding, which is specified
+in simple-panel.txt in this directory.
diff --git a/Documentation/devicetree/bindings/vendor-prefixes.txt b/Documentation/devicetree/bindings/vendor-prefixes.txt
index 55df1d444e9f..eacde76e8853 100644
--- a/Documentation/devicetree/bindings/vendor-prefixes.txt
+++ b/Documentation/devicetree/bindings/vendor-prefixes.txt
@@ -64,6 +64,7 @@ delta	Delta Electronics, Inc.
 denx	Denx Software Engineering
 digi	Digi International Inc.
 digilent	Diglent, Inc.
+dlc	DLC Display Co., Ltd.
 dlg	Dialog Semiconductor
 dlink	D-Link Corporation
 dmo	Data Modul AG
diff --git a/drivers/gpu/drm/panel/panel-simple.c b/drivers/gpu/drm/panel/panel-simple.c
index cb7dd41b5456..94da0ef15e0b 100644
--- a/drivers/gpu/drm/panel/panel-simple.c
+++ b/drivers/gpu/drm/panel/panel-simple.c
@@ -613,6 +613,31 @@ static const struct panel_desc chunghwa_claa101wb01 = {
 	},
 };
 
+/* Calculated VESA GTF mode since no datasheet is available for this panel */
+static const struct drm_display_mode dlc_dlc0700yzg_1_mode = {
+	.clock = 48960,
+	.hdisplay = 1024,
+	.hsync_start = 1024 + 40,
+	.hsync_end = 1024 + 40 + 104,
+	.htotal = 1024 + 40 + 104 + 144,
+	.vdisplay = 600,
+	.vsync_start = 600 + 1,
+	.vsync_end = 600 + 1 + 3,
+	.vtotal = 600 + 1 + 3 + 18,
+	.vrefresh = 60,
+};
+
+static const struct panel_desc dlc_dlc0700yzg_1 = {
+	.modes = &dlc_dlc0700yzg_1_mode,
+	.num_modes = 1,
+	.bpc = 6,
+	.size = {
+		.width = 154,
+		.height = 86,
+	},
+	.bus_format = MEDIA_BUS_FMT_RGB666_1X7X3_SPWG,
+};
+
 static const struct drm_display_mode edt_et057090dhu_mode = {
 	.clock = 25175,
 	.hdisplay = 640,
@@ -1129,6 +1154,9 @@ static const struct of_device_id platform_of_match[] = {
 		.compatible = "chunghwa,claa101wb01",
 		.data = &chunghwa_claa101wb01
 	}, {
+		.compatible = "dlc,dlc0700yzg-1",
+		.data = &dlc_dlc0700yzg_1,
+	}, {
 		.compatible = "edt,et057090dhu",
 		.data = &edt_et057090dhu,
 	}, {
